### DISCUSSION TITLE
Allow given instances to alias structural instances

### DISCUSSION THREAD

**Discussion by mpilquist** [scala/scala3#17812] [Feature Requests]
Consider a structural given:

```scala
object Instances:
  given listInstance: Monad[List] with Traverse[List] with ...
```

Now imagine we want to define an alias for this instance in another module:
```scala
object Functor:
  given Monad[List] with Traverse[List] = Instances.listInstance
```

This is not currently supported. Structural instances cannot be aliased via `=`. Note the following works fine though:

```scala
object Functor:
  export Instances.listInstance
```

**Comment by LPTK** [comment]
What is a structural given? I don't think I've seen such terminology before.

Can't you just write the following?:

```scala
  given Monad[List] & Traverse[List] = Instances.listInstance
```

**Comment by bishabosha** [comment]
> What is a structural given? I don't think I've seen such terminology before.

It is the terminology for a given where there is no "rhs"
e.g. `given FromString[Int] with {def fromString(s: String) = s.toInt}` or `given Ctx()` (simplified from `given Ctx with {}`)

**Comment by mpilquist** [comment]
Thanks @bishabosha. I grabbed the term from the grammar:
```
TmplDef             ::=  ...
                     |   ‘given’ GivenDef
GivenDef            ::=  [GivenSig] StructuralInstance
                     |   [GivenSig] AnnotType ‘=’ Expr
                     |   [GivenSig] AnnotType
GivenSig            ::=  [id] [DefTypeParamClause] {UsingParamClause} ‘:’
StructuralInstance  ::=  ConstrApp {‘with’ ConstrApp} ‘with’ TemplateBody
```

https://docs.scala-lang.org/scala3/reference/contextual/givens.html#syntax

**Comment by neko-kai** [comment]
@LPTK 

> What is a structural given? I don't think I've seen such terminology before.
> 
> Can't you just write the following?:
> 
> ```scala
>   given Monad[List] & Traverse[List] = Instances.listInstance
> ```

But you can't! Although you can with parentheses:

```scala
trait Monad[F[_]]
trait Traverse[F[_]]

object Instances:
  given listInstance: (Monad[List] & Traverse[List]) = new Monad[List] with Traverse[List] {}

object Functor:
  given (Monad[List] & Traverse[List]) = Instances.listInstance

@main def main = { import Functor.given ; println(summon[Monad[List]] -> summon[Traverse[List]]) }
```
https://scastie.scala-lang.org/Y006CKiITOO1hz8Hivy2fA

Essentially everything in the feature request works, but requires parentheses in a way that's inconsistent with the rest of the grammar – you don't need parentheses for an equivalent `def` or `val` definition.
Parentheses could be justified because `with` in given starts a template definition. But `&` does not start a definition, so there's no particular reason the above code should require parentheses around the type, I think.

**Comment by mpilquist** [comment]
@neko-kai Nice! I had tried the intersection but hadn't thought to use parens!
